src/value/de.rs

```diff
@@ -41,22 +41,19 @@ impl<'de> Deserialize<'de> for Value {
             fn visit_i64<E>(self, i: i64) -> Result<Value, E>
                 where E: SError,
             {
-                Ok(Value::I64(i))
+                Ok(Value::Number(Number::from(i)))
```

Comment:
FYI, reading this again I can't help but think I'd write `Value::Number(i.into())`